dynamiqs/integrators/apis/mcsolve.py

```diff
@@ -0,0 +1,272 @@
+    $\ket{\psi(t)}$ at time $t$, starting from an initial state $\ket{\psi_0}$,
+    according to the Schrödinger equation with non-Hermitian Hamiltonian ($\hbar=1$)
+    $$
+        \frac{\dd\ket{\psi(t)}}{\dt}
+        = -i [H(t) -\frac{i}{2} \sum_{k=1}^{N} L_{k}^{\dagger} (t)L_{k}(t)] \ket{\psi(t)},
+    $$
+    where $H(t)$ is the system's Hamiltonian at time $t$ and $\{L_k(t)\}$ is a
+    collection of jump operators at time $t$. We then extract the norm of the state
+    at the final time, and take this as a lower bound for random numbers sampled
+    for other trajectories such that they all experience at least one jump.
+
+    Quote: Time-dependent Hamiltonian or jump operators
+        If the Hamiltonian or the jump operators depend on time, they can be converted
+        to time-arrays using [`dq.constant()`](/python_api/time_array/constant.html),
+        [`dq.pwc()`](/python_api/time_array/pwc.html),
+        [`dq.modulated()`](/python_api/time_array/modulated.html), or
+        [`dq.timecallable()`](/python_api/time_array/timecallable.html).
+
+    Quote: Running multiple simulations concurrently
+        The Hamiltonian `H`, the jump operators `Ls` and the
+        initial state `psi0` can be batched to solve multiple monte-carlo equations
+        concurrently. All other arguments are common to every batch.
+
+    Args:
+        H _(qarray-like or time-array of shape (bH?, n, n))_: Hamiltonian.
+        Ls _(list of qarray-like or time-array, of shape (nL, n, n))_: List of
+            jump operators.
+        psi0 _(qarray-like of shape (bpsi?, n, 1))_: Initial state.
+        tsave _(array-like of shape (nt,))_: Times at which the states and expectation
+            values are saved. The equation is solved from `tsave[0]` to `tsave[-1]`, or
+            from `t0` to `tsave[-1]` if `t0` is specified in `options`.
+        keys _(Array of shape (ntraj,))_: Total number of jump trajectories to
+            simulate, not including the no-jump trajectory. Defaults to a list of keys
+            of length 10.
+        exp_ops _(list of qarray-like, of shape (nE, n, n), optional)_: List of
+            operators for which the expectation value is computed.
+        solver: Solver for the integration. Defaults to
+            [`dq.solver.Tsit5()`](/python_api/solver/Tsit5.html).
+        root_finder: Root finder passed to dx.diffeqsolve() to find the exact time an
+            event occurs. Can be `None`, in which case the root finding functionality
+            is not utilized. It is recommended to pass a root finder (such as the
+            default Newton root finder) so that event times are not determined by the
+            integration step sizes in diffeqsolve. However there are cases where the
+            root finding can fail, causing the whole simulation to fail. Passing `None`
+            for `root_finder` can alleviate the issue in these cases.
```

Comment:
We should also refer to the `diffrax` documentation with a link somewhere inside this description.